modcc/solvers.cpp

```diff
@@ -265,11 +270,81 @@ void SparseSolverVisitor::visit(AssignmentExpression *e) {
     ++deq_index_;
 }
 
+void SparseSolverVisitor::visit(ConserveExpression *e) {
+    if (A_.empty()) {
+        unsigned n = dvars_.size();
+        A_ = symge::sym_matrix(n, n);
+    }
+
+    conserve_ = true;
+
+    auto loc = e->location();
+    scope_ptr scope = e->scope();
+
+    int row_idx = -1;
+
+    auto& l = e->lhs()->is_stoich()->terms().front();
+    for (int j = 0; j < (int)dvars_.size(); ++j) {
+        auto ident = l->is_stoich_term()->ident()->is_identifier();
+        if (ident) {
+            if (ident->name() == dvars_[j]) {
+                if (j > row_idx) {
+                    row_idx = j;
+                    break;
+                }
+            }
+        }
+    }
+
+    if (row_idx != -1) {
+        A_[row_idx].clear();
+
+        for (unsigned j = 0; j < dvars_.size(); ++j) {
+            expression_ptr expr;
+
+            for (auto &l: e->lhs()->is_stoich()->terms()) {
```

Comment:
can this be replaced with `find_if`/`find`?

Reply:
Yes, I will update it